### ISSUE TITLE
Crash After 30 hours - Problem in SDK Release 4.3.1 -> (LoadStoreAlignment) (IDFGH-6107)

### ISSUE THREAD

**Issue by filzek** [espressif/esp-idf#7785]
HI, 

SDK RELEASE V4.3.1
ESP32: Wrover-b with 8MB flash and 8MB spiram.

We are debuging the update from SDK 4.2 to 4.3.1 and after all ports needed we are facing a SKD issue:

After running for almost 29 hours it crashes.

We found no memory leak in our application in internal memory or extenal spiram.

SENTINEL - ******************** EVERY 10 SECONDS **************************
SENTINEL - DATE AND TIME START: [Tue Oct 26 05:37:19 2021]
SENTINEL - DATE AND TIME NOW  : [Wed Oct 27 10:35:05 2021]
SENTINEL - HAS PASSED: [28] hours [57] minutes [102913] seconds
SENTINEL - *********************************************************
developer_Utils - showMemoryRAMStatus =======================================================
developer_Utils - showMemoryRAMStatus The current date/time is: Wed Oct 27 10:35:05 2021
developer_Utils - showMemoryRAMStatus There is 15 hardwares ports created
developer_Utils - showMemoryRAMStatus FreeHeapSize              => 3629099 bytes
developer_Utils - showMemoryRAMStatus Internal Heap Siz         => 26843 bytes
developer_Utils - showMemoryRAMStatus =======================================================
Guru Meditation Error: Core  1 panic'ed (LoadStoreAlignment). Exception was unhandled.

Core  1 register dump:
PC      : 0x400990e9  PS      : 0x00060533  A0      : 0x8009692c  A1      : 0x3ffd4c20
0x400990e9: compare_and_set_native at d:\developer\esp-idf\dev\producao\build/../../../components/esp_hw_support/include/soc/compare_set.h:33
 (inlined by) spinlock_acquire at d:\developer\esp-idf\dev\producao\build/../../../components/esp_hw_support/include/soc/spinlock.h:103
 (inlined by) vPortCPUAcquireMutex at d:\developer\esp-idf\dev\producao\build/../../../components/freertos/port/xtensa/include/freertos/portmacro.h:158
 (inlined by) vPortEnterCritical at D:/developer/esp-idf/components/freertos/port/xtensa/port.c:448

A2      : 0x202d208d  A3      : 0x0000cdcd  A4      : 0x00060523  A5      : 0x00060520
A6      : 0x0000abab  A7      : 0xb33fffff  A8      : 0x0000abab  A9      : 0x3ffd4c40
A10     : 0x3f800ca4  A11     : 0x00000001  A12     : 0x00000018  A13     : 0x00001800
A14     : 0x0000abab  A15     : 0xb33fffff  SAR     : 0x00000020  EXCCAUSE: 0x00000009
EXCVADDR: 0x202d208d  LBEG    : 0x40091c7c  LEND    : 0x40091c98  LCOUNT  : 0x00000000
0x40091c7c: memcpy at /builds/idf/crosstool-NG/.build/HOST-x86_64-w64-mingw32/xtensa-esp32-elf/src/newlib/newlib/libc/machine/xtensa/memcpy.S:175

0x40091c98: memcpy at /builds/idf/crosstool-NG/.build/HOST-x86_64-w64-mingw32/xtensa-esp32-elf/src/newlib/newlib/libc/machine/xtensa/memcpy.S:197

Backtrace:0x400990e6:0x3ffd4c20 0x40096929:0x3ffd4c50 0x4015bb89:0x3ffd4c90 0x4014ca69:0x3ffd4cc0 0x4014f2cf:0x3ffd4ce0 0x40157b65:0x3ffd4d00 0x40157c1e:0x3ffd4d50 0x40154d9c:0x3ffd4d80 0x40154e15:0x3ffd4dc0 0x40154f25:0x3ffd4de0 0x4015deb7:0x3ffd4e20 0x4014c9c1:0x3ffd4e40
0x400990e6: compare_and_set_native at d:\developer\esp-idf\dev\producao\build/../../../components/esp_hw_support/include/soc/compare_set.h:33
 (inlined by) spinlock_acquire at d:\developer\esp-idf\dev\producao\build/../../../components/esp_hw_support/include/soc/spinlock.h:103
 (inlined by) vPortCPUAcquireMutex at d:\developer\esp-idf\dev\producao\build/../../../components/freertos/port/xtensa/include/freertos/portmacro.h:158
 (inlined by) vPortEnterCritical at D:/developer/esp-idf/components/freertos/port/xtensa/port.c:448

0x40096929: xQueueGenericSend at D:/developer/esp-idf/components/freertos/queue.c:790

0x4015bb89: sys_mbox_trypost at D:/developer/esp-idf/components/lwip/port/esp32/freertos/sys_arch.c:264

0x4014ca69: tcpip_try_callback at D:/developer/esp-idf/components/lwip/lwip/src/api/tcpip.c:366

0x4014f2cf: netif_loop_output at D:/developer/esp-idf/components/lwip/lwip/src/core/netif.c:1190

0x40157b65: ip4_output_if_opt_src at D:/developer/esp-idf/components/lwip/lwip/src/core/ipv4/ip4.c:1074

0x40157c1e: ip4_output_if_src at D:/developer/esp-idf/components/lwip/lwip/src/core/ipv4/ip4.c:907

0x40154d9c: udp_sendto_if_src at D:/developer/esp-idf/components/lwip/lwip/src/core/udp.c:906 (discriminator 4)

0x40154e15: udp_sendto_if at D:/developer/esp-idf/components/lwip/lwip/src/core/udp.c:705

0x40154f25: udp_sendto at D:/developer/esp-idf/components/lwip/lwip/src/core/udp.c:612

0x4015deb7: lwip_netconn_do_send at D:/developer/esp-idf/components/lwip/lwip/src/api/api_msg.c:1640

0x4014c9c1: tcpip_thread_handle_msg at D:/developer/esp-idf/components/lwip/lwip/src/api/tcpip.c:208
 (inlined by) tcpip_thread at D:/developer/esp-idf/components/lwip/lwip/src/api/tcpip.c:154

SDK is broken for some bug, not sure where, but you guys can find and fix it, maybe related to others Guru Meditation Error: Core  1 panic'ed (LoadStoreAlignment). Exception was unhandled. reports.

**Comment by Alvin1Zhang** [comment]
@filzek Thanks for reporting, would you please help share if any updates for the issue? Thanks.

**Comment by filzek** [comment]
We have found out the problem, it is memory realted but not a memory leak at all.

The sdk version V4.2.2 uses LESS memory than the current V4.3.1, sometimes it pull 7kb more and this makes the esp32 runs out of memory so it crashes, but doesnt know why it happen at around 30 hours.

We have tuned the code to use a little bit less memory and disable some other code and it is running well.

So, the V4.3.1 break some realtime code when something use more memory, but the memory is relased correct after those use maybe a bluetooth/wifi/other non user memory area.

So as the application has left a 6k block only and a 4k aligned block its crash around 30 hours, but when the memory left is around 10000 bytes and 8000 bytes in a block it doenst crash.

Another finding, the V4.2.2 doesnt lock the core 1 / core 2 interrupt, and the version V4.3.1 after and while (not sequention, so totaly random but happen at least about 8 times in an hour) it miss the GPIO interrupt act, the GPIO is running a zerocross in 60hz, so it happen 120 times in a second, about 8333 microssecond, and when it miss it stop the GPIO for about few microssends but less than a second, so it show a WIPE in the GPIO status (failure), maybe sometime a GPIO is blocking a core action, so it is very important to take a look closer as it is similar to some patches in the DEV brunch, maybe it is important to release a patch release V4.3.2 with some bugfixes.
